compiler/AST/primitive.cpp

```diff
@@ -702,11 +702,12 @@ initPrimitive() {
   prim_def(PRIM_REQUIRE, "require", returnInfoVoid, false, false);
 }
 
-Map<const char*, VarSymbol*> memDescsMap;
+static Map<const char*, VarSymbol*> memDescsMap;
+static Map<int, VarSymbol*> memDescsNodeMap;
```

Comment:
So here is the trade-off:

* hash on IDs => somebody is surprised => they read the comment on memDescsNodeMap (which I will add) and figure it out

* hash on Type* => a minuscule chance of a bug when we possibly delete a Type* possibly in the future => misleading information in memLeaks output => either doesn't matter or takes hours? days? to figure out

Your call.